content/using_ec2_spot_instances_with_eks/jenkins/increasing_resilience.md

```diff
@@ -0,0 +1,61 @@
+---
+title: "Increasing resilience"
+date: 2018-08-07T08:30:11-07:00
+weight: 60
+---
+
+
+#### Challenge
+
+As oppose to the stateless web application that we deployed previously in the workshop, Jenkins jobs are not fault-tolerant, meaning that an EC2 Spot interruption on an instance that is running a job, will cause the job to fail. 
+Is there anything we can do to decrease the chances of having Spot interruptions, when EC2 needs the capacity back?\
+Hint: read or skim through the following [article] (https://aws.amazon.com/blogs/compute/introducing-the-capacity-optimized-allocation-strategy-for-amazon-ec2-spot-instances/)
+
+{{%expand "Click here for the answer" %}}
+
+Previously, we used the lowest-price Spot allocation strategy, by selecting a large number of `spotInstancePools` in our nodegroups configuration, in order to deploy our applications on a diversified set of the cheapest EC2 Spot instance types. We did this for the purpose of decreasing the blast radius in case Spot Interruptions occur - which will only terminate a small portion of our instances, due to the diversification. However, since Jenkins job oriented workloads are not fault-tolerant and an EC2 Spot interruption would cause the build job to fail, we can choose the capacity-optimized allocation strategy which will provision Spot Instances for us from the capacity pools that have the lowest chances of being interrupted. This way, we increase the chances of successfully completing our Jenkins jobs when running on Spot Instances.
```

Comment:
bear in mind readers should be reading this after having read the capacity-optimized article, so I'm hoping this would be clear. open to any suggestions on how to simplify - I did my best :)